web/packages/teleterm/src/ui/TopBar/NavigationMenu/NavigationMenu.tsx

```diff
@@ -18,75 +18,102 @@ import React, { useState, useRef } from 'react';
 import styled from 'styled-components';
 
 import Popover from 'design/Popover';
-import { MoreVert, OpenBox, Add } from 'design/Icon';
-import { Box, Text, Flex } from 'design';
+import { MoreVert, OpenBox, Add, Config } from 'design/Icon';
 
 import { useAppContext } from 'teleterm/ui/appContextProvider';
-import { DocumentsService } from 'teleterm/ui/services/workspacesService';
 import { TopBarButton } from 'teleterm/ui/TopBar/TopBarButton';
-import { RootClusterUri } from 'teleterm/ui/uri';
-
-import { useIdentity } from '../Identity/useIdentity';
+import { IAppContext } from 'teleterm/ui/types';
+import { Cluster } from 'teleterm/services/tshd/types';
 
 import { NavigationItem } from './NavigationItem';
 
-function getNavigationItems(
-  documentsService: DocumentsService,
-  clusterUri: RootClusterUri
-): {
-  title: string;
-  Icon: JSX.Element;
-  onNavigate: () => void;
-}[] {
+function useNavigationItems(): (
+  | {
+      type?: 'normal';
```

Comment:
I see where this is coming from but it's a little bit weird as the value isn't actually used anywhere. If we want to use `type`/`kind` field, I think we should either commit to it fully or use something else instead.

If we want to make defining the nav items here easier, we could let go of the `type` field and use `'separator'` as the type for the separator instead of `{ type: 'separator' }`.

Reply:
It seemed a little weird to me too, but I didn't have a better idea. `separator` string is cool.